pkg/collector/corechecks/containerlifecycle/queue.go

```diff
@@ -0,0 +1,103 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package containerlifecycle
+
+import (
+	"errors"
+	"sync"
+
+	model "github.com/DataDog/agent-payload/v5/contlcycle"
+)
+
+type queue struct {
+	chunkSize int
+	data      []model.EventsPayload
+	sync.RWMutex
```

Comment:
can we reuse a `sync.Map` here?

Reply:
Our main data is `[]model.EventsPayload` not a map though, plus  `sync.Map` requires casting on every operation which is not ideal here imho.

Reply:
A yes indeed